File: backend/app/schemas/analytics.py
from typing import List, Optional, Dict, Any
from pydantic import BaseModel, Field

class ForecastPoint(BaseModel):
    date: str
    predicted_quantity: float
    p10_lower: float
    p90_upper: float
    actual_quantity: Optional[float] = None

class MedicineForecast(BaseModel):
    medicine_id: str
    medicine_name: str
    facility_id: str
    facility_name: str
    horizon_days: int
    current_stock: int
    predicted_burn_rate_daily: float
    days_until_stockout: float
    predicted_stockout_date: Optional[str]
    forecast_points: List[ForecastPoint]
    historical_points: List[ForecastPoint]
    confidence_score: float
    anomaly_detected: bool = False
    anomaly_reason: Optional[str] = None

class HourlyBedForecastPoint(BaseModel):
    timestamp: str
    general_beds_demand: int
    icu_beds_demand: int
    emergency_beds_demand: int
    surge_risk_score: float

class BedDemandForecast(BaseModel):
    facility_id: str
    facility_name: str
    horizon_hours: int = 72
    points: List[HourlyBedForecastPoint]
    peak_general_demand: int
    peak_icu_demand: int
    icu_saturation_risk_pct: float

class ConsumptionTrend(BaseModel):
    category: str
    total_consumed_30d: int
    growth_rate_pct: float
    seasonal_multiplier: float

class AnalyticsOverview(BaseModel):
    total_consumed_ytd: int
    predicted_demand_next_14d: int
    model_accuracy_pct: float
    active_anomalies_count: int
    monitored_medicines_count: int
    categories_trend: List[ConsumptionTrend]

File: frontend/src/types/analytics.ts
export interface ForecastPoint {
  date: string
  predicted_quantity: number
  p10_lower: number
  p90_upper: number
  actual_quantity?: number | null
}

export interface MedicineForecast {
  medicine_id: string
  medicine_name: string
  facility_id: string
  facility_name: string
  horizon_days: number
  current_stock: number
  predicted_burn_rate_daily: number
  days_until_stockout: number
  predicted_stockout_date?: string | null
  forecast_points: ForecastPoint[]
  historical_points: ForecastPoint[]
  confidence_score: number
  anomaly_detected: boolean
  anomaly_reason?: string | null
}

export interface HourlyBedForecastPoint {
  timestamp: string
  general_beds_demand: number
  icu_beds_demand: number
  emergency_beds_demand: number
  surge_risk_score: number
}

export interface BedDemandForecast {
  facility_id: string
  facility_name: string
  horizon_hours: number
  points: HourlyBedForecastPoint[]
  peak_general_demand: number
  peak_icu_demand: number
  icu_saturation_risk_pct: number
}

export interface ConsumptionTrend {
  category: string
  total_consumed_30d: number
  growth_rate_pct: number
  seasonal_multiplier: number
}

export interface AnalyticsOverview {
  total_consumed_ytd: number
  predicted_demand_next_14d: number
  model_accuracy_pct: number
  active_anomalies_count: number
  monitored_medicines_count: number
  categories_trend: ConsumptionTrend[]
}
